File: SWK5/uebung01/Basic/Calc.cs
using System.Reflection;

[assembly:AssemblyVersion("1.0.0.0")]

public class Calc
{
    protected double sum = 0;
    protected int n = 0;

    public void Add(double number)
    {
        sum += number;
        n++;
    }

    public double GetSum()
    {
        return sum;
    }
}

File: SWK5/uebung01/managedcodeManagedTypes/Calc.cpp
#include "Calc.h"

using namespace System;

Calc::Calc() {
	sum = 0;
	n   = 0;
}

Calc::~Calc() {
	Console::WriteLine("~MCalc\n");
}

Calc::!Calc() {
	Console::WriteLine("!MCalc\n");
}

void Calc::Add(double number) {
	sum += number;
	n++;
}

double Calc::GetSum() {
	return sum;
}
